libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -378,12 +378,17 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 			if (_variable.isConstructorParameter())
 				errorString += " for constructor parameter";
 			else if (_variable.isCallableOrCatchParameter())
-				errorString +=
-					" for " +
-					string(_variable.isReturnParameter() ? "return " : "") +
-					"parameter in" +
-					string(_variable.isExternalCallableParameter() ? " external" : "") +
-					" function";
+			{
+							string visibility;
+							if (auto const* callable = dynamic_cast<CallableDeclaration const*>(_variable.scope()))
+								visibility = Declaration::visibilityToString(callable->visibility());
+							errorString +=
+							" for " +
+							string(_variable.isReturnParameter() ? "return " : "") +
+							"parameter in" +
+							string(visibility.empty() ? visibility : " " + visibility) +
+							" function";
```

Comment:
We could use our `joinHumanReadable()` to make this more concise (needs `#include <libsolutil/StringUtils.h`). This removes the need to worry about including or omitting the spaces between components:
```suggestion
							string parameterKind = _variable.isReturnParameter() ? "return parameter"s : "parameter"s;
							errorString += " " + joinHumanReadable({"for", parameterKind, "in", visibility, "function"}, " ");
```

Reply:
I tried using joinHumanReadable() but it could not find the function even with the header included. I have opted to for the second approach. 

There are still whitespace issues that I am fat fingering. ( I am missing how to obtain the correct styling completely....)

Reply:
Maybe that's because this file does not use the namespace where the function is defined yet. Please try `util::joinHumanReadable()` or add `using namespace solidity::util;` at the top of the file.

Reply:
The whitespace issue seems to be that you're indenting with spaces while we use tabs for .cpp files.

I see that the `chk_coding_style` job detects this but does not print any message about what is wrong. We should change that at some point.

Reply:
got you, let me try again real fast